File: CodeGeneratorMVC/oldClasses/ObjectExtensions.vb
Imports System.Windows.Forms
Imports System.Runtime.CompilerServices

Public Module ObjectExtensions
    <Extension()> _
    Public Sub Flag(ByVal x As DataGridViewCell)
        x.Tag = True
    End Sub
    Public Sub UnFlag(ByVal x As DataGridViewCell)
        x.Tag = False
    End Sub
    <Extension()> _
    Public Function isFlagged(ByVal x As DataGridViewCell) As Boolean
        Return CBool(x.Tag)
    End Function
    <Extension()> _
    Public Function Count(ByVal st As String, ch As Char) As Integer
        Dim cnt As Integer = 0
        For Each c As Char In st
            If c = ch Then cnt += 1
        Next
        Return cnt
    End Function
End Module

File: CodeGeneratorMVC/Models/ObjectExtensions.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Globalization;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Runtime.CompilerServices;
using System.Security;
using System.Text;
using System.Threading.Tasks;
using Microsoft.VisualBasic;

public static class ObjectExtensions {
    //public static void Flag(this DataGridViewCell x) {
    //    x.Tag = true;
    //}
    //public static void UnFlag(DataGridViewCell x) {
    //    x.Tag = false;
    //}
    //public static bool isFlagged(this DataGridViewCell x) {
    //    return System.Convert.ToBoolean(x.Tag);
    //}
    public static int Count(this string st, char ch) {
        int cnt = 0;
        foreach (char c in st) {
            if (c == ch)
                cnt += 1;
        }
        return cnt;
    }
}
